### PR TITLE
PR#1

### PR DESCRIPTION
**Pull request by VTerenya**
Check my solution, please)

### CODE REVIEW

File: calculator.go [hunk 1]
```diff
@@ -1 +1,38 @@
+package main
 
+import (
+	"flag"
+	"fmt"
+	"volume_calculator/Shapes"
+)
+
+func volumeOfShape(shape string, parameters...float64) float64 {
```
Comment (rtemb) [inline calculator.go:9]: Basically you don't need that function. You could init shape in the `if` clause from main() and before and before calling fmt.Println() call  `sphere.Volume()

File: calculator.go [hunk 2]
```diff
@@ -1 +1,38 @@
...
+	widthPtr := flag.Float64("width", -1.0, "a float64")
+	hieghtPtr := flag.Float64("hieght", -1.0, "a float64")
+	if shape == "sphere" {
+		fmt.Println(volumeOfShape(shape, *radiurPtr))
+	} else if shape == "pyramid" {
+		fmt.Println(volumeOfShape(shape, *lengthPtr, *widthPtr, *hieghtPtr))
+	} else {
+		fmt.Println(volumeOfShape(shape, *hieghtPtr, *radiurPtr))
+	}
+	flag.Parse()
```
Comment (rtemb) [inline calculator.go:36]: I guess that should go before `if`

File: calculator.go [hunk 3]
```diff
@@ -1 +1,38 @@
...
+}
+
+func main() {
+	var shape string
+	flag.StringVar(&shape, "shape", "string", "a string")
+	radiurPtr := flag.Float64("radius", -1.0, "a float64")
+	lengthPtr := flag.Float64("length", -1.0, "a float64")
+	widthPtr := flag.Float64("width", -1.0, "a float64")
+	hieghtPtr := flag.Float64("hieght", -1.0, "a float64")
+	if shape == "sphere" {
```
Comment (rtemb) [inline calculator.go:29]: I'd propose to use switch here with default section which will stop the program with an error. Currently you have kind of bug: if you pass unknown shape as the argument, the app will always calculate cylinder volume.

File: calculator.go [hunk 4]
```diff
@@ -1 +1,38 @@
...
+		fmt.Println(volumeOfShape(shape, *radiurPtr))
+	} else if shape == "pyramid" {
+		fmt.Println(volumeOfShape(shape, *lengthPtr, *widthPtr, *hieghtPtr))
+	} else {
+		fmt.Println(volumeOfShape(shape, *hieghtPtr, *radiurPtr))
+	}
+	flag.Parse()
+	fmt.Println(shape)
```
Comment (rtemb) [inline calculator.go:37]: To implement a pretty formatting you can implement standard `Stringer` interface

File: calculator.go (hunk 1 above)
Comment (VTerenya) [reply calculator.go:9]: I just want to check exactly this function in unit test. I can't do this for visibility?

File: calculator.go (hunk 4 above)
Comment (VTerenya) [reply calculator.go:37]: I should always implement Stringer? In this case, I thought that it's not important, or it's mistake?

File: calculator.go (hunk 2 above)
Comment (VTerenya) [reply calculator.go:36]: Okay, i'll fix it.

File: calculator.go (hunk 3 above)
Comment (VTerenya) [reply calculator.go:29]: Yes, i agree with you. I'll remake this code.

File: Shapes/shapes.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package Shapes
```
[the remark this answers is not in the document]
Comment (VTerenya) [reply Shapes/shapes.go:1]: Sorry, of course)

File: Shapes/shapes.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"math"
+)
+
+type Sphere struct{
+	Radius float64
```
[the remark this answers is not in the document]
Comment (VTerenya) [reply Shapes/shapes.go:8]: Yes, I understood.

File: calculator.go [hunk 7]
```diff
@@ -1 +1,38 @@
...
+	} else {
+		cylinder := Shapes.Cylinder{Hieght: parameters[0], Radius: parameters[1]}
+		return cylinder.Volume()
+	}
+}
+
+func main() {
+	var shape string
+	flag.StringVar(&shape, "shape", "string", "a string")
+	radiurPtr := flag.Float64("radius", -1.0, "a float64")
```
[the remark this answers is not in the document]
Comment (VTerenya) [reply calculator.go:25]: I'll fix it, thank you.

File: calculator.go [hunk 8]
```diff
@@ -1 +1,38 @@
...
+	"fmt"
+	"volume_calculator/Shapes"
+)
+
+func volumeOfShape(shape string, parameters...float64) float64 {
+	if shape == "sphere" {
+		sphere := Shapes.Sphere{Radius: parameters[0]}
+		return sphere.Volume()
+	} else if shape == "pyramid" {
+		pyramid := Shapes.Pyramid{Length: parameters[0], Width: parameters[1], Hieght: parameters[2]}
```
[the remark this answers is not in the document]
Comment (VTerenya) [reply calculator.go:14]: Yes, it must have)

File: calculator.go [hunk 9]
```diff
@@ -1 +1,40 @@
...
+		sphere := shapes.Sphere{Radius: radius}
+		return sphere.Volume()
+	case "pyramid":
+		pyramid := shapes.Pyramid{Length: length, Width: width, Hieght: hieght}
+		return pyramid.Volume()
+	case "cylinder":
+		cylinder := shapes.Cylinder{Hieght: hieght, Radius: radius}
+		return cylinder.Volume()
+	default:
+		panic("Panic... I don't know this shape!")
```
Comment (rtemb) [inline calculator.go:26]: It will be more correctly to use println to print the message + os.Exit(1) to stop the program correctly

**Review by rtemb — COMMENTED** [review]
In general looks good. Only minor changes left

File: calculator.go (hunk 9 above)
Comment (VTerenya) [reply calculator.go]: I fixed it, thank you.

File: calculator.go (hunk 4 above)
Comment (rtemb) [reply calculator.go]: It's not a mistake. It just an option, You could create your custom formatting when you print a shape.

File: calculator.go [hunk 10]
```diff
@@ -1 +1,42 @@
...
+	}
+}
+
+func main() {
+	var shape string
+	flag.StringVar(&shape, "shape", "string", "a string")
+	radius := flag.Float64("radius", -1.0, "a float64")
+	length := flag.Float64("length", -1.0, "a float64")
+	width := flag.Float64("width", -1.0, "a float64")
+	hieght := flag.Float64("hieght", -1.0, "a float64")
```
Comment (rtemb) [inline calculator.go:38]: you still may lose accuracy when you parsing args to the float types.

**Review by rtemb — APPROVED** [review]

**Pull request closed without merging**